### DISCUSSION TITLE
Support for MonoDB

### DISCUSSION THREAD

**Discussion by simondeeley** [dunglas/symfony-docker#144] [Ideas]
Are there any plans to incorporate MongoDB as an alternative to MariaDB for those of us where our business cases require MonoDB over a traditional SQL-style database?

**Comment by maxhelias** [comment]
We integrate none of them 😄 
So everyone is free to choose the one they want.

**Reply by simondeeley** [reply]
Perhaps a better way of wording it would be, where is the best place to add PHP extensions to the Dockerfile?  MongoDB PHP extension usually "prefers" PECL to install.

**Reply by maxhelias** [reply]
This project is made to be self-configuring thanks to symfony/flex and these recipes.
If you take a look at the Dockerfile here : https://github.com/dunglas/symfony-docker/blob/0abc295461d2b03ea807e638282ea748e368cb39/Dockerfile#L96 
There are markers so that Flex can add or remove elements in this file.
For the docker-compose, there is no need since we know how to parse yaml file.

Now look the panther recipes here : https://github.com/symfony/recipes/blob/master/symfony/panther/1.0/manifest.json
When you add this dependency in your project (`composer require symfony/panther`), thanks to the symfony/flex, the lines you see in the "dockerfile" section will be added enter the markers or remove if you remove the dependency.

For DB, an opinon choice was taken in the doctrine/doctrine-bundle dependency to install PostgreSQL https://github.com/symfony/recipes/blob/master/doctrine/doctrine-bundle/2.3/manifest.json

But a new recipes can be put on another dependency if you use something else. Feel free to do the PR on the symfony/recipe or symfony/recipe-contrib repository

**Reply by simondeeley** [reply]
Thanks for the clarification, and this is what I thought would happen.

Yet when I try to install the Flex recipe for `doctrine/mongodb-odm-bundle` it errors saying minimum requirements are not met.

Specifically the PHP version and the PHP mongoldb extension.  This led me to believe that I would need to add the mongodb extension manually and then install the ODM bundle in the "traditional" composer way, allowing for the more recent release supporting PHP > 8.0

**Reply by maxhelias** [reply]
Oh I see! It's probably related to the constraints of composer. Try your "require" with the flag `--ignore-platform-reqs`

**Comment by maxhelias** [comment]
You can use the `maker-bundle` with the command `make:docker:database` to generate a database service where you can choose between MySQL / MariaDB / Postgres : https://github.com/symfony/maker-bundle/pull/640
